### PR TITLE
add kosaraju algorithm in cpp

### PR DESCRIPTION
**Pull request by pragyajaiswa05l**
Fixes #2246 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- add kosaraju algorithm in cpp
- 

#### Languages Used:
- c++
- 

#### Files Added:
- kosaraju_algorithm.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: kosaraju_algorithm/kosaraju_algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+6 5
+
+OUTPUT
+6 5
+2 3 4 1 */
+
+
+
+
+
```
Comment (asha15) [inline kosaraju_algorithm/kosaraju_algorithm.cpp:109]: remove these lines

File: kosaraju_algorithm/kosaraju_algorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        cout << endl;
+        delete component;
+        it++;
+    }
+
+    delete components;
+    delete [] edges;
+    delete [] edgesT;
+}
```
Comment (asha15) [inline kosaraju_algorithm/kosaraju_algorithm.cpp:90]: keep a line space after this

File: kosaraju_algorithm/kosaraju_algorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        unordered_set<int>* component = new unordered_set<int>();
+        dfs2(edgesT, element, component, visited);
+        output->insert(component);
+    }
+    return output;
+}
+
+int main() {
+    int n;                                                 //no.of vertices
```
Comment (asha15) [inline kosaraju_algorithm/kosaraju_algorithm.cpp:56]: remove the extra spaces before the comment

File: kosaraju_algorithm/kosaraju_algorithm.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    cout<<"Enter the no. of vertices : "<<endl;
+    cin >> n;
+    vector<int>* edges = new vector<int>[n];   //using a adjacency list to implement
+    vector<int>* edgesT = new vector<int>[n];             //transpose of a graph
+    int m;                                                // no.of edges
```
Comment (asha15) [inline kosaraju_algorithm/kosaraju_algorithm.cpp:61]: remove the extra spaces before the comment

File: kosaraju_algorithm/kosaraju_algorithm.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    for (int i = 0; i < edges[start].size(); i++) {
+        int adjacent = edges[start][i];
+        if (visited.count(adjacent) == 0) {
+            dfs(edges, adjacent, visited, finishStack);
+        }
+    }
+    finishStack.push(start);
+}
+
+ //dfs of a transpose graph
```
Comment (asha15) [inline kosaraju_algorithm/kosaraju_algorithm.cpp:20]: remove the space at the beginning

**Comment by pragyajaiswa05l** [comment]
@asha15  mam please review it i have made a changes  which were required

**Pull request merged**